### PR TITLE
Magic Ladder

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
Jacob's ladder item that will create a linked ladder on every accessible Z level lined up with one another for exciting!! crew adventures into the void/lavaland/etc

It's in the necropolis chest. I consolidated some of the various modkits into the same drop rolls because the list is getting kind of long

:cl:
add: Added Jacob's ladder to the necropolis chest
/:cl:

@yoshmaster thought of this

### CODE REVIEW

**Comment by ExcessiveUseOfCobblestone** [comment]
hahahahaha

**Comment by PraiseRatvar** [comment]
zEELEVELS

**Comment by DoctoreMad** [comment]
What's with the esoteric references in Lavaland?
Is this officially hell?

**Comment by KorPhaeron** [comment]
¯\_(ツ)_/¯

**Comment by kevinz000** [comment]
@KorPhaeron goddamnit what if they ladder into the AI core/armory?!!

**Comment by ExcessiveUseOfCobblestone** [comment]
emergent gameplay

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 1]
```diff
@@ -69,8 +75,7 @@
 		if(27)
 			new /obj/item/borg/upgrade/modkit/lifesteal(src)
 			new /obj/item/bedsheet/cult(src)
-		if(28)
```
Comment (ChangelingRain) [inline code/modules/mining/lavaland/necropolis_chests.dm]: update the rand!

Comment (ExcessiveUseOfCobblestone) [reply code/modules/mining/lavaland/necropolis_chests.dm]: can we just talk about how cancer this thing looks

**Comment by AnonymousNow** [comment]
Radical.

**Comment by Tokiko1** [comment]
I thought it was @Yoshmaster.

**Comment by KorPhaeron** [comment]
wtf

**Comment by KorPhaeron** [comment]
How embarrassing!

**Comment by BlueNothing** [comment]
What is this a reference to?

**Comment by DoctoreMad** [comment]
Consult your local search engine lazybones.

**Comment by DoctoreMad** [comment]
Here's a hint though: Its not the film.

**Comment by BlueNothing** [comment]
Binding of Isaac, or the biblical symbol?

**Comment by DoctoreMad** [comment]
Checking the BoI wiki Jacob's Ladder doesn't seem to be a actual ladder while the biblical symbol is about a ladder form Earth to Heaven. Now it doesn't take too much to discover it's probably the biblical symbol you mentioned.

**Comment by jughu** [comment]
What if its the ladder from the legend of zelda that allows you to cross 1 tile of lava

**Comment by Supermichael777** [comment]
that's a bridge in the Japanese wording i think

**Comment by Jordie0608** [comment]
Ok but why

**Comment by KorPhaeron** [comment]
I think the game needs more non weapon items that lead to silly hijinks

**Comment by Cyberboss** [comment]
This has got to be  the most immulsion breaking thing ever. Also it will lead to transit space which will fuck the climber. :-1:

**Comment by KorPhaeron** [comment]
It does not lead to transit

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 2]
```diff
@@ -563,7 +568,27 @@
...
+	name = "jacob's ladder"
+	desc = "A celestial ladder that violates the laws of physics, but respects the coordinate plane."
+	icon = 'icons/obj/structures.dmi'
+	icon_state = "ladder00"
+
+/obj/item/jacobs_ladder/attack_self(mob/user)
+	var/turf/T = get_turf(src)
+	var/ladder_x = T.x
+	var/ladder_y = T.y
+	to_chat(user, "You unfold the ladder. It extends much farther than you were expecting.")
```
Comment (Cyberboss) [inline code/modules/mining/lavaland/necropolis_chests.dm]: notice span

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 3]
```diff
@@ -563,7 +568,27 @@
...
+	for(var/i in 1 to world.maxz)
+		if(i == ZLEVEL_CENTCOM || i == ZLEVEL_TRANSIT)
+			continue
+		new /obj/structure/ladder/unbreakable/jacob(ladder_x, ladder_y, i)
+	qdel(src)
 
+/obj/structure/ladder/unbreakable/jacob
+	name = "jacob's ladder"
+	desc = "An indestructible celestial ladder that violates the laws of physics, but respects the coordinate plane."
```
Comment (Cyberboss) [inline code/modules/mining/lavaland/necropolis_chests.dm]: coordinate plane is not an IC friendly term

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 4]
```diff
@@ -563,7 +568,27 @@
 	..()
 
 
+/obj/item/jacobs_ladder
+	name = "jacob's ladder"
+	desc = "A celestial ladder that violates the laws of physics, but respects the coordinate plane."
```
Comment (Cyberboss) [inline code/modules/mining/lavaland/necropolis_chests.dm]: coordinate plane is not an IC friendly term

File: code/modules/mining/lavaland/necropolis_chests.dm (hunk 1 above)
Comment (Cyberboss) [inline code/modules/mining/lavaland/necropolis_chests.dm]: update the rand

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/mining/lavaland/necropolis_chests.dm (hunk 3 above)
Comment (DoctoreMad) [reply code/modules/mining/lavaland/necropolis_chests.dm]: You said that twice

File: code/modules/mining/lavaland/necropolis_chests.dm (hunk 4 above)
Comment (KorPhaeron) [reply code/modules/mining/lavaland/necropolis_chests.dm]: Its a player visible/in universe thing and is what is used for the GPS/suit sensors/telesci etc, I'll switch to a different term if anyone has one though

Comment (ExcessiveUseOfCobblestone) [reply code/modules/mining/lavaland/necropolis_chests.dm]: "quadrant"

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**